pvoutput/grid_search.py

```diff
@@ -0,0 +1,763 @@
+        default=25.0,
+        help="Specify the radial search limit around each grid point in " "kilometers.",
+    )
+    parser.add_argument(
+        "--local-crs-epsg",
+        metavar="<EPSG>",
+        dest="local_crs_epsg",
+        action="store",
+        type=int,
+        required=False,
+        default=4087,
+        help="Optionally provide the EPSG code of a local co-ordinate Reference "
+        "System (CRS) for improved accuracy. e.g. set to 27700 (OSGB36 / "
+        "British National Grid) if searching the British Isles.",
+    )
+    parser.add_argument(
+        "--cache-dir",
+        metavar="</path/to/dir>",
+        dest="cache_dir",
+        action="store",
+        type=str,
+        required=False,
+        default=None,
+        help="Specify a directory to use for caching downloaded boundary files.",
+    )
+    parser.add_argument(
+        "--show",
+        dest="show",
+        action="store_true",
+        required=False,
+        help="Set this flag to show a plot of the grid.",
+    )
+    parser.add_argument(
+        "-o",
+        "--outfile",
+        metavar="<path>",
+        dest="outfile",
+        action="store",
+        type=str,
+        required=False,
+        help="Specify a filename to save the grid to.",
+    )
+    options = parser.parse_args()
+
+    def handle_options(options):
+        """Validate command line args and pre-process where necessary."""
+        if options.bbox is not None:
+            options.bbox = list(map(lambda x: float(x.strip()), options.bbox.split(",")))
+        if options.radial_clip is not None:
+            options.radial_clip = list(
+                map(lambda x: float(x.strip()), options.radial_clip.split(","))
+            )
+        if options.cache_dir is not None:
+            if not os.path.isdir(options.cache_dir):
+                logging.error(f"The cache_dir '{options.cache_dir}' does not exist.")
+                raise ValueError(f"The cache_dir '{options.cache_dir}' does not exist.")
+        if options.countries:
+            options.countries = list(map(lambda x: str(x.strip()), options.countries.split(",")))
+            earth = NaturalEarth(options.cache_dir)
+            _, countries = earth.get_hires_world_boundaries()
+            for country in options.countries:
+                if country not in countries:
+                    logging.error(f"The country '{country}' is invalid.")
+                    raise ValueError(f"The country '{country}' is invalid.")
+        if options.outfile is not None and os.path.exists(options.outfile):
+            check = query_yes_no(
+                f"The output file '{options.outfile}' already exists, results "
+                "will be overwritten, do you want to continue?",
+                "no",
+            )
+            if check is False:
+                print("Quitting...")
+                sys.exit(0)
+        return options
+
+    return handle_options(options)
+
+
+def main():
```

Comment:
Would it be possible to move this to a script folder? then it snot in the main package?

Reply:
Will do before merging the PR. Eventually it might be more appropriate to move the generic(ish) `grid_search.py` code to a separate repo and make it pip-installable, but we can look at that later.

Reply:
Thanks a good idea